ckanext/s3filestore/controller.py

```diff
@@ -63,14 +63,18 @@ def resource_download(self, id, resource_id, filename=None):
                 # We are using redirect to minio's resource public URL
                 s3 = upload.get_s3_session()
                 client = s3.client(service_name='s3', endpoint_url=host_name)
+
+                # check whether the object exists in S3
+                client.head_object(Bucket=bucket_name, Key=key_path)
```

Comment:
Are we not needing to store the result?

Reply:
At this point, no. We could maybe do something with it in future, like directly serving small files and redirecting for larger ones, but what we really need is to trigger the 404 error handling.